mep3_navigation/src/distance_angle/distance_angle_regulator.cpp

```diff
@@ -119,18 +143,22 @@ void DistanceAngleRegulator::odometry_callback(const nav_msgs::msg::Odometry::Sh
   const double distance_increment_angle = std::atan2(delta_y, delta_x);
```

Comment:
Ovo je malo cudan nacin za trunutnog stanja u odnosu na cilj. Je l mozemo da izvucemo to bez prev_robot_x_ i prev_robot_y_? Trebalo bi da moze samo na osnovu cilja i trenutne pozicije/orjentacije.

Reply:
To nije vezano za cilj. Potrebno je iz /odom gde imamo dekartov koordinatni sistem da pretvorimo u "polarni" distance-angle sistem. Da bi se to postiglo racuna se koliku je distancu robot presao u odnosu na prethodnu tacku i na osnovu ugla duzi koja spaja te dve tacke odlucuje se da li se povecao "distance" ili smanjio.

Dakle ovo samo /odom pretvori u dve koordinate: distance i angle.